src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete copy[key];
+        }
+        break;
+
+      case 'clear':
+        for (const del in copy) {
+          delete copy[del];
+        }
+        break;
+
+      default:
+        break;
```

Comment:
Better to throw an error in default case for example:
`throw new Error('Unknown action type')`